impala/hiveserver2.py

```diff
@@ -1040,7 +1049,7 @@ def __init__(self, trowset, expect_more_rows, schema, convert_types=True):
 
             # STRING columns are read as binary and decoded here to be able to handle
             # non-valid utf-8 strings in Python 3.
-            if six.PY3:
+            if six.PY3 and convert_strings_to_unicode:
```

Comment:
The case when convert_strings_to_unicode is False but convert_types is True also needs to be consider - the code in _convert_values probably expects Unicode when converting DATE/TIMESTAMP/DECIMAL types.

Reply:
Makes sense, please check out new design.